### PR TITLE
Sync master branch to 2.2.4

### PR DESCRIPTION
**Pull request by bigspider**

### CODE REVIEW

File: .github/workflows/codeql-workflow.yml [hunk 1]
```diff
@@ -17,8 +17,6 @@ jobs:
     strategy: 
       matrix:
         include:
-          - SDK: "$NANOS_SDK"
```
Comment (b0l0k) [inline .github/workflows/codeql-workflow.yml:20]: Is it on purpose?

Comment (bigspider) [reply .github/workflows/codeql-workflow.yml:20]: Yes, Nano S won't have future versions and the main branches are no longer compatible with Nano S starting from this version.

The last Nano S version is kept on its own `nanos` branch.

File: src/handler/register_wallet.c [hunk 2]
```diff
@@ -123,44 +130,39 @@ void handler_register_wallet(dispatcher_context_t *dc, uint8_t protocol_version)
...
+                                                        cosigner_index,
+                                                        (uint8_t *) keys_info[cosigner_index],
+                                                        MAX_POLICY_KEY_INFO_LEN);
 
-        if (pubkey_info_len < 0) {
+        if (key_info_len < 0) {
             SEND_SW(dc, SW_INCORRECT_DATA);
-            ui_post_processing_confirm_wallet_registration(dc, false);
             return;
         }
```
Comment (b0l0k) [inline src/handler/register_wallet.c:156]: I'd add a control on `key_info_len > MAX_POLICY_KEY_INFO_LEN + 1` even though I assume that `call_get_merkle_leaf_element` has already this control.

**Review by b0l0k — APPROVED** [review]
LGTM.
Happy to see that dropping the support of Nano S get rid of many memory trick.

**Pull request merged**